src/components/Register.jsx

```diff
@@ -0,0 +1,27 @@
+import React, { useState } from 'react';
+
+import Input from './Input';
+import AuthLayout from './AuthLayout';
+
+const Register = () => {
+    const [registerDetails, setRegisterDetails] = useState({
+        username: '',
+        password: '',
+        income: ''
+    });
+
+    const onRegisterChange = (newState) => {
+        setRegisterDetails({ ...registerDetails, ...newState });
+    };
+
+    return (
+        <AuthLayout name='register'>
+            <Input id="username" onChange={onRegisterChange} value={registerDetails.username} label="username"/>
+            <Input id="income" onChange={onRegisterChange} value={registerDetails.income} label="income"/>
```

Comment:
I would suggest rather capturing the user's income as part of the capture incomes and expenses process.